x-pack/plugins/lens/public/editor_frame_service/editor_frame/state_helpers.ts

```diff
@@ -97,15 +105,33 @@ export async function persistedStateToExpression(
 
   const datasourceLayers = createDatasourceLayers(datasources, datasourceStates);
 
-  return buildExpression({
-    title,
-    description,
+  const datasourceId = getActiveDatasourceIdFromDoc(doc);
+  if (datasourceId == null) {
+    return {
+      ast: null,
+      errors: [{ shortMessage: '', longMessage: getMissingCurrentDatasource() }],
+    };
+  }
+  const validationResult = validateDatasourceAndVisualization(
+    datasources[datasourceId],
+    datasourceStates[datasourceId].state,
     visualization,
     visualizationState,
-    datasourceMap: datasources,
-    datasourceStates,
-    datasourceLayers,
-  });
+    { datasourceLayers } as FramePublicAPI
```

Comment:
This is a little scary - can be tighten the types here a bit? I went through our current consumers and there's one place using something besides `datasourceLayers` - the xy visualization will check out `activeData`.

Can we change the type of the second parameter of `getConfiguration` to `Pick<FramePublicAPI, 'datasourceLayers' | 'activeData'>`? This should fix the type cast and we want to move away from the kitchen sink api anyway.

Reply:
Sure 👍

Reply:
There's also `availablePalettes` which is required somewhere. I couldn't find a way to drop it for the embeddable so for now I'm loading them.

Reply:
Managed to isolate it and remove `availablePalettes` as dependency as it was unused